internal/btf/btf.go

```diff
@@ -484,14 +494,65 @@ func (s *Spec) Program(name string, length uint64) (*Program, error) {
 	return &Program{s, length, funcInfos, lineInfos, relos}, nil
 }
 
-// FindType searches for a type with a specific name.
+// TypeByID returns the BTF Type with the given type ID.
+//
+// Returns an error wrapping ErrNotFound if a Type with the given ID
+// does not exist in the Spec.
+func (s *Spec) TypeByID(id TypeID) (Type, error) {
+	if int(id) > len(s.types) {
+		return nil, fmt.Errorf("type ID %d: %w", id, ErrNotFound)
+	}
+	return s.types[id], nil
+}
+
+// AnyTypesByName returns a list of BTF Types with the given name. If name
+// contains a struct flavor suffix (e.g. 'thread_struct___v46'), an exact match
+// will be performed. Otherwise, returns all types that carry the given name,
+// including all struct flavors.
 //
-// Called T a type that satisfies Type, typ must be a non-nil **T.
-// On success, the address of the found type will be copied in typ.
+// Multiple Types of the same name can exist in case multiple 'struct flavors'
+// are described in BTF info. These are used to deal with changes in kernel
+// data structures.
+//
+// If the BTF blob describes multiple compilation units like vmlinux, multiple
+// Types with the same name and kind can exist, but might not describe the same
+// data structure.
+//
+// Returns an error wrapping ErrNotFound if no matching Type exists in the Spec.
+func (s *Spec) AnyTypesByName(name string) ([]Type, error) {
+	en := essentialName(name)
+
+	// Named types are indexed by essential name.
+	types, ok := s.namedTypes[en]
+	if !ok {
+		return nil, fmt.Errorf("type name %s: %w", name, ErrNotFound)
+	}
+
+	// If name contains a flavor suffix, require an exact match.
+	if en != name {
```

Comment:
What about this instead?

```suggestion
	if en == name {
		return types, nil
	}
```

I think we should probably also copy `types` to avoid the user modifying `namedTypes`.

Reply:
`return types, nil` is already the fallthrough case at the end of the function. This clause here implements matching exact struct flavors.